Duplicati/Library/Backend/Telegram/TelegramBackend.cs

```diff
@@ -0,0 +1,473 @@
+            var userDialogsSlice = absDialogs as TLDialogsSlice;
+            TLVector<TLAbsChat> absChats;
+
+            if (userDialogs != null)
+            {
+                absChats = userDialogs.Chats;
+            }
+            else
+            {
+                absChats = userDialogsSlice.Chats;
+            }
+
+            var channel = (TLChannel)absChats.FirstOrDefault(chat => chat is TLChannel tlChannel && tlChannel.Title == m_channelName);
+            m_channelCache = channel;
+            return channel;
+        }
+
+        private void EnsureChannelCreated()
+        {
+            var channel = GetChannel();
+            if (channel == null)
+            {
+                var newGroup = new TLRequestCreateChannel
+                {
+                    Broadcast = false,
+                    Megagroup = false,
+                    Title = m_channelName,
+                    About = string.Empty
+                };
+
+                EnsureConnected();
+                m_telegramClient.SendRequestAsync<object>(newGroup).GetAwaiter().GetResult();
+            }
+        }
+
+        private void Authenticate()
+        {
+            EnsureConnected();
+
+            if (IsAuthenticated())
+            {
+                return;
+            }
+
+            try
+            {
+                var phoneCodeHash = m_sessionStore.GetPhoneHash(m_phoneNumber);
+                if (phoneCodeHash == null)
+                {
+                    EnsureConnected();
+                    phoneCodeHash = m_telegramClient.SendCodeRequestAsync(m_phoneNumber).GetAwaiter().GetResult();
+                    m_sessionStore.SetPhoneHash(m_phoneNumber, phoneCodeHash);
+                    m_telegramClient.Session.Save();
+
+                    throw new UserInformationException(Strings.NoOrWrongAuthCodeError, nameof(Strings.NoOrWrongAuthCodeError));
+                }
+
+                m_telegramClient.MakeAuthAsync(m_phoneNumber, phoneCodeHash, m_authCode).GetAwaiter().GetResult();
+            }
+            catch (CloudPasswordNeededException)
+            {
+                if (string.IsNullOrEmpty(m_password))
+                {
+                    m_telegramClient.Session.Save();
+                    throw new UserInformationException(Strings.NoPasswordError, nameof(Strings.NoPasswordError));
+                }
+
+                EnsureConnected();
+                var passwordSetting = m_telegramClient.GetPasswordSetting().GetAwaiter().GetResult();
+                m_telegramClient.MakeAuthWithPasswordAsync(passwordSetting, m_password).GetAwaiter().GetResult();
+            }
+
+            m_telegramClient.Session.Save();
+        }
+
+        private void EnsureConnected()
+        {
+            var isConnected = false;
+
+            while (isConnected == false)
+            {
+                try
+                {
+                    isConnected = m_telegramClient.ConnectAsync().Wait(TimeSpan.FromSeconds(5));
```

Comment:
Make sure Wait() will not wrap FloodException, if so handle the specific case, or use another timeout logic